src/SessionManager.ts

```diff
@@ -0,0 +1,158 @@
+/**
+ * Copyright 2020 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import IStorage from "./storage/IStorage";
+import Session from "./Session";
+import { EventEmitter } from "events";
+import AuthFetcher from "./AuthFetcher";
+import getAuthFetcherWithDependencies from "./dependencies";
+import { detectEnvironment } from "./util/EnvironmentDetector";
+import ISessionInfo from "./sessionInfo/ISessionInfo";
+
+export interface ISessionManagerOptions {
+  secureStorage?: IStorage;
+  insecureStorage?: IStorage;
+}
+
+export class SessionManager extends EventEmitter {
+  private authFetcher: AuthFetcher;
+  private sessionRecords: Record<
+    string,
+    { session: Session; logoutCallback: () => unknown }
+  > = {};
+  private isInitialized = false;
+  private handledIncomingRedirect = false;
+
+  constructor(options: ISessionManagerOptions) {
+    super();
+    this.authFetcher = getAuthFetcherWithDependencies({
+      secureStorage: options.secureStorage,
+      insecureStorage: options.insecureStorage
+    });
+  }
+
+  private async init(): Promise<void> {
+    if (!this.isInitialized) {
+      const env = detectEnvironment();
+      if (env === "server" || env === "react-native") {
+        try {
+          await this.handleIncomingRedirect(window.location.href);
+        } catch (err) {
+          // Do nothing
```

Comment:
I generally feel swallowed exceptions warrant a lot more explanation than just 'Do nothing'. Here for instance, I don't know why it's totally Ok to completely ignore all redirect handling problems (i.e. it doesn't 'feel' Ok!).

Reply:
Fixed in 00f8fd7216a630a4ba3b649e9104ea2bc333a4d5